apps/accounts/tests/test_views.py

```diff
@@ -0,0 +1,81 @@
+from django.core.urlresolvers import reverse
+from django.test.utils import override_settings
+
+import mock
+
+from rest_framework.test import APITestCase
+from waffle import Switch
+
+from users.models import UserProfile
+
+FXA_CONFIG = {'some': 'stuff', 'that is': 'needed'}
+
+
+class override_switch(object):
+    def __init__(self, name, active):
+        self.name = name
+        self.active = active
+
+    def __enter__(self):
+        Switch.objects.create(name=self.name, active=self.active)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        Switch.objects.filter(name=self.name).delete()
+
+
+class TestFxALoginWaffle(APITestCase):
+
+    @property
+    def url(self):
+        return reverse('accounts.login')
+
+    def test_404_when_waffle_is_off(self):
+        with override_switch('fxa-auth', active=False):
```

Comment:
Can't you just use [self.create_switch()](https://github.com/mozilla/olympia/blob/master/apps/amo/tests/__init__.py#L472) ? Since it's a Django test case, the DB will be rolled back afterwards.